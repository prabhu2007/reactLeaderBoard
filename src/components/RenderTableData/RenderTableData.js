import { React } from 'react';

export const RenderTableData = (props) => {
  const { leaderBoard } = props;

  if (!leaderBoard) {
    return <div>No Data</div>;
  }

  return leaderBoard.map((item, index) => {
    const { player, score } = item;

    return (
      <tr className="player-list-item" key={index}>
        <th>{index + 1}</th>
        <td>{player}</td>
        <td>{score}</td>
      </tr>
    );
  });
};
