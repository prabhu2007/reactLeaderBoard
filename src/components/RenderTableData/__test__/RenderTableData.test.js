import React from 'react';
import { shallow } from 'enzyme';
import { RenderTableData } from '../RenderTableData';

describe('RenderTableData component', () => {
  it('should render correctly with no props', () => {
    const renderTableDataComponent = shallow(<RenderTableData />);

    expect(renderTableDataComponent).toMatchSnapshot();
  });

  it('should render correctly with given array', () => {
    const leaderBoard = [
      { name: 'John', score: '200' },
      { name: 'Tim', score: '100' },
    ];
    const render_table_component = shallow(
      <RenderTableData leaderBoard={leaderBoard} />,
    );

    expect(render_table_component).toMatchSnapshot();
  });
});
