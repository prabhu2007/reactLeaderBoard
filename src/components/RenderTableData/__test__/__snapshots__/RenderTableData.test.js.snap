// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderTableData component should render banner text correctly with given array 1`] = `ShallowWrapper {}`;

exports[`RenderTableData component should render correctly with given array 1`] = `ShallowWrapper {}`;

exports[`RenderTableData component should render correctly with no props 1`] = `ShallowWrapper {}`;
